src/mock/quic_p2p/tests.rs

```diff
@@ -596,14 +593,20 @@ impl Agent {
     }
 
     // Expect `Event::UnsentUserMessage` with the given recipient address and content.
-    fn expect_unsent_message(&self, dst_addr: &SocketAddr, expected_msg: &NetworkBytes) {
-        let (actual_addr, actual_msg) = assert_match!(
+    fn expect_unsent_message(
```

Comment:
I am not 100% sure the test cover the msg_id for SentUserMessage, and not UnsentUserMessage

Reply:
I'm not sure what you mean - could you elaborate a bit?

Reply:
Most of the msg_id seem to be `0` in the test, and I did not really see additional testing to cover all the new behaviours introduced in the mock by this PR.

So my question is did we cover our new code with test that validate the new behaviours.

Reply:
No, I think we didn't. Fair point.